src/main/java/ui/components/pickers/MilestonePickerState.java

```diff
@@ -273,6 +276,18 @@ private void addMatchingMilestones(List<PickerMilestone> milestonesToBePopulated
                 .findFirst();
     }
 
+    /**
+     * Returns the list of milestones that the user is allowed to select.
+     * A milestone is allowed for selection if it is not an existing milestone.
+     *
+     * @param milestones
+     */
+    private List<PickerMilestone> getSelectableMilestones(List<PickerMilestone> milestones) {
```

Comment:
This method should be `static` method in `PickerMilestone` (like the ones you removed), so that this method is reusable by other classes that involve `PickerMilestone` but not `PickerMilestoneState`. Also, this method is relevant to `PickerMilestone` rather than this class.